shape.go

```diff
@@ -372,36 +412,50 @@ type Shape interface {
 	ShapeInheritor
 	ShapeBaser
 	ShapeChecker
-	// Clones the shape and its children and points to specified base shape.
+	// ShapeCloner Clones the shape and its children and points to specified base shape.
 	ShapeCloner
 	ShapeValidator
 
 	yamlNodesUnmarshaller
 	fmt.Stringer
+	IsScalar() bool
 }
 
-// identifyShapeType identifies the type of the shape.
-func identifyShapeType(shapeFacets []*yaml.Node) string {
-	var t = TypeString
+// identifyShapeType identifies the type of the shape by facets.
+func identifyShapeType(shapeFacets []*yaml.Node) (string, error) {
+	var t = ""
 	for i := 0; i != len(shapeFacets); i += 2 {
 		node := shapeFacets[i]
-		if _, isString := SetOfStringFacets[node.Value]; isString {
-			t = TypeString
-		} else if _, isInteger := SetOfNumberFacets[node.Value]; isInteger {
-			t = TypeInteger
-			break
-		} else if _, isFile := SetOfFileFacets[node.Value]; isFile {
-			t = TypeFile
-			break // File has a unique facet
-		} else if _, isObj := SetOfObjectFacets[node.Value]; isObj {
-			t = TypeObject
-			break
-		} else if _, isArray := SetOfArrayFacets[node.Value]; isArray {
-			t = TypeArray
-			break
+		var ft string
+		var ok bool
+		switch node.Value {
+		case FacetMinLength, FacetMaxLength, FacetPattern:
+			ok = true
+			ft = TypeString
+		case FacetMinimum, FacetMaximum, FacetMultipleOf:
+			ok = true
+			ft = TypeNumber
+		case FacetMinItems, FacetMaxItems, FacetUniqueItems, FacetItems:
+			ok = true
+			ft = TypeArray
+		case FacetMinProperties, FacetMaxProperties, FacetAdditionalProperties, FacetProperties, FacetDiscriminator:
+			ok = true
+			ft = TypeObject
+		case FacetFileTypes:
+			ok = true
+			ft = TypeFile
+		}
+		if ok {
+			if t != "" && ft != t {
+				return "", fmt.Errorf("detected types by facets are not equal: %s and %s", t, ft)
```

Comment:
This will not work as expected and this function must not produce any errors.
```
#%RAML 1.0 DataType

minLength: 10
fileTypes: [ "*/*" ]
```

Step by step:
1. `ft = TypeString` and `t = TypeString` because minLength is present.
2. `ft = TypeFile` and `t = TypeString` because fileTypes is present.
3. `t != "" && ft != t` condition fails.